### PR TITLE
created OverflowingChipsWithTooltip component

### LINKED ISSUE #1086: Improve display of numerous relations - Implement 'show more' Overlay
# Current Behavior:
Today, when there are too many relations, they are displayed without consideration of available space, leading to a cluttered appearance.
### Example issue on tasks:

https://user-images.githubusercontent.com/102394218/258555670-69f2c5c9-add6-45fc-ad88-cdc4c8b6374f.mp4

# Expected Behavior: 
The desired behavior is to apply a consistent display format for all multiple relations across the application.
### Proposed Solution:
1.	Limiting Width Div:
o	When displaying relations, constrain them within a maximum width div.
o	The div should not exceed the length of a single line.
2.	Displaying Maximum Elements:
o	Show the maximum number of relations that fit within the div.
3.	Adding 'Show More' Indicator:
o	If the div is full and there are additional non-displayed relations, add a '+n' indicator, representing the number of hidden relations.
4.	'Show More' Interaction:
o	Make the '+n' indicator clickable.
o	When clicked, open an overlay that expands to the div's width and accommodates the content to display all the hidden relations.

### Example desired behavior on tasks:

https://user-images.githubusercontent.com/102394218/258555611-b7dc316e-b028-4b35-b414-5bb7a3e21b7e.mp4

By implementing these changes, we aim to improve the display of numerous relations, ensuring a neat and organized presentation, and providing users with the option to access all relations via the 'Show More' overlay.

### Proposed implementation plan

```[tasklist]
### Tasks
- [ ] Introduce a @/ui/OverflowingChipsWithTooltip component that take children as an argument and applying a width: 100% to itself
- [ ] Add a useEffect to compute container size in js, compute estimated chip size based on text and icon
- [ ] Add a +n button with remaining items
- [ ] On click on the +n button, display a overflow div with the same width as it parent
- [ ] On click outside, close the overflow div
- [ ] Use this new component in Activity Right Panel, and on Task inbox
```

### PR DESCRIPTION
**Pull request by mohitshende**
closes #1086

### CODE REVIEW

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      entries.forEach((entry) => {
+        if (entry.target === containerRef.current) {
+          setContainerWidth(entry.contentRect.width);
+        }
+      });
+    });
+
+    resizeObserver.observe(containerRef.current);
+
+    return () => {
```
Comment (emilienchvt) [inline front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:69]: Why is this return statement required ? Nit: commented code

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    refs: [containerRef],
+    callback: () => {
+      setIsOverflowOpen(false);
+    },
+  });
+
+  useEffect(() => {
+    const items = React.Children.toArray(children);
+
+    const itemWidth = 14 + 60; //icon width + assumed text width
```
Comment (emilienchvt) [inline front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:92]: The apporach sounds good 👍  Out of curiosity, how did you come to that expected size ?

**Review by emilienchvt — COMMENTED** [review]
Thanks for getting started on this one @mohitshende, way to go 🚀 

I tested the feature and found a few cases, leading to some issues, for example: 

In the task list when the text is long and a lot of entities selected:
Screenshot 2023-08-23 at 17 41 01

In the task drawer, when overflow is not active:
Screenshot 2023-08-23 at 17 42 00

Same task, overflow active:
Screenshot 2023-08-23 at 17 42 37

Did you test these cases ?

**Comment by charlesBochet** [comment]
@mohitshende Thanks a lot for the help!
Any update on this one?
It seems that this is not working properly yet, feel free to ping me here if you need more help

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx (hunk 1 above)
Comment (mohitshende) [reply front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:69]: It is recommended to clear the side effects during unmounting of a component, that's why I wrote this part. I will clear the comment.

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx (hunk 2 above)
Comment (mohitshende) [reply front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:92]: 14px is the width of logo that I saw in chips component, regarding text size I couldn't find a way to calculate it for every children. If you have any suggestion for it, do let me know.

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    const totalVisibleItems = items.slice(0, containerWidth / itemWidth);
```
[the remark this answers is not in the document]
Comment (mohitshende) [reply front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:93]: Yes, I will make changes accordingly.

**Comment by mohitshende** [comment]
> @mohitshende Thanks a lot for the help! Any update on this one? It seems that this is not working properly yet, feel free to ping me here if you need more help

Yes, I have identified issues and I am trying to fix them. The major problem that I am facing is that chips are overflowing out of task row, which shouldn't be the case. And I need to give some explicit width to this component so that it can calculate visible chips with the logic I wrote. If you have any suggestions on how should we do it. Please let me know.

**Comment by charlesBochet** [comment]
@mohitshende I'll take a look tomorrow

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+    resizeObserver.observe(containerRef.current);
+
+    return () => {
+      // setContainerWidth(0);
+      resizeObserver.disconnect();
+    };
+  }, [isNavOpen, isRightDrawerExpanded]);
+
+  const handleOverflowButtonClick = (
```
Comment (lucasbordeau) [inline front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:75]: We use function instead of const as a convention.

**Review by lucasbordeau — COMMENTED** [review]
Please could you look at the implementation already done in OverflowingTextWithTooltip.tsx and see if it could be made the same for this component ? It is more simple, but I don't know if it covers this use case properly.

File: front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx (hunk 4 above)
Comment (mohitshende) [reply front/src/modules/ui/overflowing-chips-with-tooltip/OverflowingChipsWithTooltip.tsx:75]: @lucasbordeau I will make changes accordingly.

**Comment by lucasbordeau** [comment]
Hi @mohitshende I'm closing this PR due to inactivity, please re-open it as soon as you have more time for this.

**Pull request closed without merging**